lamesearch.py

```diff
@@ -1,20 +1,27 @@
 #!/usr/bin/env python3
 
-import os, collections
+import os, collections, re
 
 DIR = "../sample"
 filenames = sorted(os.listdir(DIR))
 del filenames[250:]
 titles = []
 
 index_by_term = collections.defaultdict(set)
+translation_table = dict.fromkeys(map(ord,'!{}[]<>.,;:'), None)
 
 for i, filename in enumerate(filenames):
     print(filename)
     with open(os.path.join(DIR, filename)) as f:
         title = f.readline().strip()
         titles.append(title)
-        tokens = f.read().lower().split()
+
+        text = f.read().lower()
+
+        match_criteria = re.compile('[^a-zA-Z0-9\\\/]|_')
```

Comment:
In Python, it's preferred to use a raw string for a regexp: `r'[^a-zA-Z0-9\/]|_'`

(better because you don't have to double backslashes, which are super common in regexps)